packages/gitbook/src/lib/visitors.ts

```diff
@@ -221,6 +231,27 @@ function parseVisitorQueryParamValue(value: string): ClaimPrimitive {
     return value;
 }
 
+/**
+ * Returns to cookie response to use in order to persist visitor params that were passed to the URL.
+ */
+function getResponseCookieForVisitorParams(
+    visitorParamsClaims: Record<string, ClaimPrimitive>
+): ResponseCookie | undefined {
+    if (Object.keys(visitorParamsClaims).length === 0) {
+        return undefined;
+    }
+
+    return {
+        name: VISITOR_UNSIGNED_CLAIMS_PREFIX,
+        value: JSON.stringify(visitorParamsClaims),
+        options: {
+            httpOnly: true,
```

Comment:
Why `httpOnly`? It means only the server can read it and not client side scripts